tests/test_ipynb.py

```diff
@@ -0,0 +1,93 @@
+            "temp/nbs/test_2.ipynb::Cell 3": 4,
+            "temp/nbs/test_3.ipynb::Cell 0": 5,
+            "temp/nbs/test_3.ipynb::Cell 1": 1,
+            "temp/nbs/test_3.ipynb::Cell 2": 1,
+            "temp/nbs/test_3.ipynb::Cell 3": 2,
+            "temp/nbs/test_3.ipynb::Cell 4": 1,
+            "temp/nbs/test_4.ipynb::Cell 0": 1,
+            "temp/nbs/test_4.ipynb::Cell 1": 1,
+            "temp/nbs/test_4.ipynb::Cell 2": 3,
+        }
+        items = [item(x) for x in durations.keys()]
+        algo = Algorithms[algo_name].value
+        g1, g2, g3 = algo(splits=3, items=items, durations=durations)
+
+        assert g1.selected == [
+            item(nodeid="temp/nbs/test_1.ipynb::Cell 0"),
+            item(nodeid="temp/nbs/test_1.ipynb::Cell 1"),
+            item(nodeid="temp/nbs/test_1.ipynb::Cell 2"),
+            item(nodeid="temp/nbs/test_2.ipynb::Cell 0"),
+            item(nodeid="temp/nbs/test_2.ipynb::Cell 1"),
+        ]
+        assert g2.selected == [
+            item(nodeid="temp/nbs/test_2.ipynb::Cell 2"),
+            item(nodeid="temp/nbs/test_2.ipynb::Cell 3"),
+            item(nodeid="temp/nbs/test_3.ipynb::Cell 0"),
+            item(nodeid="temp/nbs/test_3.ipynb::Cell 1"),
+        ]
+        assert g3.selected == [
+            item(nodeid="temp/nbs/test_3.ipynb::Cell 2"),
+            item(nodeid="temp/nbs/test_3.ipynb::Cell 3"),
+            item(nodeid="temp/nbs/test_3.ipynb::Cell 4"),
+            item(nodeid="temp/nbs/test_4.ipynb::Cell 0"),
+            item(nodeid="temp/nbs/test_4.ipynb::Cell 1"),
+            item(nodeid="temp/nbs/test_4.ipynb::Cell 2"),
+        ]
+
+        _reorganize_broken_up_ipynbs(g1, items)
+        assert g1.selected == [
+            item(nodeid="temp/nbs/test_1.ipynb::Cell 0"),
+            item(nodeid="temp/nbs/test_1.ipynb::Cell 1"),
+            item(nodeid="temp/nbs/test_1.ipynb::Cell 2"),
+            item(nodeid="temp/nbs/test_2.ipynb::Cell 0"),
+            item(nodeid="temp/nbs/test_2.ipynb::Cell 1"),
+            item(nodeid="temp/nbs/test_2.ipynb::Cell 2"),
+            item(nodeid="temp/nbs/test_2.ipynb::Cell 3"),
+        ]
+
+        _reorganize_broken_up_ipynbs(g2, items)
+        assert g2.selected == [
+            item(nodeid="temp/nbs/test_3.ipynb::Cell 0"),
+            item(nodeid="temp/nbs/test_3.ipynb::Cell 1"),
+            item(nodeid="temp/nbs/test_3.ipynb::Cell 2"),
+            item(nodeid="temp/nbs/test_3.ipynb::Cell 3"),
+            item(nodeid="temp/nbs/test_3.ipynb::Cell 4"),
+        ]
+
+        _reorganize_broken_up_ipynbs(g3, items)
+        assert g3.selected == [
+            item(nodeid="temp/nbs/test_4.ipynb::Cell 0"),
+            item(nodeid="temp/nbs/test_4.ipynb::Cell 1"),
+            item(nodeid="temp/nbs/test_4.ipynb::Cell 2"),
+        ]
+
+
+if __name__ == "__main__":
+    import warnings
+
+    warnings.filterwarnings("ignore")
+
+    import pytest
+
+    args = "test_ipynb.py"
+    retcode = pytest.main(args.split())
```

Comment:
Now the CI seems to be failing as it's not reaching the required code coverage limit. These are dead lines so if you remove these, the CI should be green 🙂   See the GHA run [here](https://github.com/ma-sadeghi/pytest-split/runs/4495842442?check_suite_focus=true).

![Screenshot 2021-12-12 at 10 18 35](https://user-images.githubusercontent.com/30327563/145705443-a5c68376-2c62-418b-973b-5d098a578ae8.png)


You don't need any main with `pytest`. You can run it from command line with `pytest tests/test_ipynb.py`